### PR TITLE
Fix for issue #138: Add null check for device name to avoid NPE

### PR DESCRIPTION
**Pull request by y-a-n-n**

### CODE REVIEW

**Comment by hannesa2** [comment]
Does it build ? 
I tried it several minutes ago and I failed

**Comment by y-a-n-n** [comment]
I'm trying to sort out issues with the build as the gradle config hasn't been updated in a while. I've got a fork where I've merged your WorkWithAndroidStudio and added some config updates of my own. If I can get that working, I may submit a separate PR, but I'm not hopeful it'll get merged considering that PRs haven't been merged here for a while.

File: software/applications/HelloIOIO/build.gradle [hunk 1]
```diff
@@ -3,16 +3,21 @@ apply plugin: 'com.android.application'
 buildscript {
   repositories {
     mavenCentral()
+      google()
+      jcenter()
   }
 
   dependencies {
-    classpath 'com.android.tools.build:gradle:2.3.3'
+    classpath 'com.android.tools.build:gradle:4.0.0'
```
Comment (hannesa2) [inline software/applications/HelloIOIO/build.gradle:11]: Btw, it's a bad praxis to specify `com.android.tools.build:gradle:xxx` for each project/module.
It's much better to specify it just once in topmost `build.gradle`

Same to 

```
buildscript {
  repositories {
    mavenCentral()
      google()
      jcenter()
  }
```

**Comment by y-a-n-n** [comment]
Thanks for the feedback. The project structure wasn't defined by me; I'm just trying to stay consistent with the original repo.

I also didn't intend the gradle changes to make it into this PR, so I might close it and reopen two separate ones: one for the functional change and one for the build system issues (once I resolve them--currently having issues with resolving dependency via JitPack).

**Comment by y-a-n-n** [comment]
See https://github.com/ytai/ioio/pull/152

**Pull request closed without merging**